test/unit/io/sequence_file/sequence_file_output_test.cpp

```diff
@@ -475,7 +475,9 @@ void compression_by_filename_impl(test::tmp_filename & filename, std::string_vie
         buffer = std::string{std::istreambuf_iterator<char>{fi}, std::istreambuf_iterator<char>{}};
     }
 
-    EXPECT_EQ(buffer, expected);
+    // Ignore the OS flag
+    EXPECT_EQ(buffer.substr(0,9), expected.substr(0,9));
```

Comment:
Addressing the issue @eseiler mentioned that the weird OS byte is only set by the gzip format but not bzip I would remove the EXPECT equalities here from the impl() function and instead test equality in the individual tests where we can test for exact equality for bzip and "exact without the 11th byte" for gzip